provider/marathon/config.go

```diff
@@ -0,0 +1,416 @@
+		Domain       string // FIXME DEAD ?
+	}{
+		Applications: filteredApps,
+		Domain:       p.Domain, // FIXME DEAD ?
+	}
+
+	configuration, err := p.GetConfiguration("templates/marathon.tmpl", MarathonFuncMap, templateObjects)
+	if err != nil {
+		log.Errorf("failed to render Marathon configuration template: %s", err)
+	}
+	return configuration
+}
+
+func (p *Provider) applicationFilter(app marathon.Application) bool {
+	// Filter disabled application.
+	if !label.IsEnabledP(app.Labels, p.ExposedByDefault) {
+		log.Debugf("Filtering disabled Marathon application %s", app.ID)
+		return false
+	}
+
+	// Filter by constraints.
+	constraintTags := label.GetSliceStringValueP(app.Labels, label.TraefikTags)
+	if p.MarathonLBCompatibility {
+		if haGroup := label.GetStringValueP(app.Labels, labelLbCompatibilityGroup, ""); len(haGroup) > 0 {
+			constraintTags = append(constraintTags, haGroup)
+		}
+	}
+	if p.FilterMarathonConstraints && app.Constraints != nil {
+		for _, constraintParts := range *app.Constraints {
+			constraintTags = append(constraintTags, strings.Join(constraintParts, ":"))
+		}
+	}
+	if ok, failingConstraint := p.MatchConstraints(constraintTags); !ok {
+		if failingConstraint != nil {
+			log.Debugf("Filtering Marathon application %v pruned by '%v' constraint", app.ID, failingConstraint.String())
+		}
+		return false
+	}
+
+	return true
+}
+
+func (p *Provider) taskFilter(task marathon.Task, application marathon.Application) bool {
+	if task.State != string(taskStateRunning) {
+		return false
+	}
+
+	// Filter task with existing, bad health check results.
+	if application.HasHealthChecks() {
+		if task.HasHealthCheckResults() {
+			for _, healthCheck := range task.HealthCheckResults {
+				if !healthCheck.Alive {
+					log.Debugf("Filtering Marathon task %s from application %s with bad health check", task.ID, application.ID)
+					return false
+				}
+			}
+		}
+	}
+
+	if ready := p.readyChecker.Do(task, application); !ready {
+		log.Infof("Filtering unready task %s from application %s", task.ID, application.ID)
+		return false
+	}
+
+	return true
+}
+
+// getFrontendRule returns the frontend rule for the specified application, using
+// its label. If service is provided, it will look for serviceName label before generic one.
+// It returns a default one (Host) if the label is not present.
+func (p *Provider) getFrontendRule(application marathon.Application, serviceName string) string {
+	labels := getLabels(application, serviceName)
+	lblFrontendRule := getLabelName(serviceName, label.SuffixFrontendRule)
+	if value := label.GetStringValue(labels, lblFrontendRule, ""); len(value) > 0 {
+		return value
+	}
+
+	if p.MarathonLBCompatibility {
+		if value := label.GetStringValueP(application.Labels, labelLbCompatibility, ""); len(value) > 0 {
+			return "Host:" + value
+		}
+	}
+
+	if len(serviceName) > 0 {
+		return "Host:" + strings.ToLower(provider.Normalize(serviceName)) + "." + p.getSubDomain(application.ID) + "." + p.Domain
```

Comment:
Should not we use `fmt.Sprintf`?

Reply:
because it's not Java  :stuck_out_tongue_closed_eyes:

Reply:
But, IMHO, it's easier to read. :smile_cat: